File: src/examples/benchmarking/benchmarking_multiproc.go
package main

import (
	"fmt"
	"joey"
	"path/filepath"
	"runtime"
	"time"
)

func setup() string {
	_, file, _, ok := runtime.Caller(0)
	if !ok {
		panic("could not get file path.")
	}
	dir := filepath.Dir(file)
	testCsvFilePath := filepath.Join(dir, "benchmarking_data.csv")
	return testCsvFilePath
}

func main() {
	dataframe, err := joey.NewFromCsv(setup())
	if err != nil {
		panic(err)
	}
	// Convert to int
	for _, column := range *dataframe.Columns() {
		column.Convert("int")
	}

	fmt.Println("----- Benchmarking Adding Columns -----")
	// Single Proc
	start_time := time.Now()
	joey.N_PROC = 1
	dataframe.Column("charge").Add(*dataframe.Column("walltime"))
	duration := time.Since(start_time)
	fmt.Printf("Single Proc Add Duration: %v seconds.\n", duration)

	// Multi Proc
	start_time = time.Now()
	joey.N_PROC = 8
	dataframe.Column("charge").Add(*dataframe.Column("walltime"))
	duration = time.Since(start_time)
	fmt.Printf("Multi Proc Add Duration: %v seconds.\n\n", duration)

	fmt.Println("----- Benchmarking Subtracting Columns -----")
	// Subtracting a new column from another
	// Single Proc
	start_time = time.Now()
	joey.N_PROC = 1
	dataframe.Column("charge").Subtract(*dataframe.Column("walltime"))
	duration = time.Since(start_time)
	fmt.Printf("Single Proc Subtract Duration: %v seconds.\n", duration)

	// Multi Proc
	start_time = time.Now()
	joey.N_PROC = 8
	dataframe.Column("charge").Subtract(*dataframe.Column("walltime"))
	duration = time.Since(start_time)
	fmt.Printf("Multi Proc Subtract Duration: %v seconds.\n\n", duration)

	fmt.Println("----- Benchmarking Multiplying Columns -----")
	// Subtracting a new column from another
	// Single Proc
	start_time = time.Now()
	joey.N_PROC = 1
	dataframe.Column("charge").Multiply(*dataframe.Column("walltime"))
	duration = time.Since(start_time)
	fmt.Printf("Single Proc Multiply Duration: %v seconds.\n", duration)

	// Multi Proc
	start_time = time.Now()
	joey.N_PROC = 8
	dataframe.Column("charge").Multiply(*dataframe.Column("walltime"))
	duration = time.Since(start_time)
	fmt.Printf("Multi Proc Multiply Duration: %v seconds.\n\n", duration)

}

File: src/examples/benchmarking/benchmarking_multiproc.py
import pandas as pd
import time, os

if __name__ == "__main__":
    dir = os.path.dirname(os.path.abspath(__file__))
    dataframe = pd.read_csv(os.path.join(dir, "benchmarking_data.csv"))

    print("----- Benchmarking Adding Columns -----")
    start_time = time.time()
    dataframe["charge"] = dataframe["charge"] + dataframe["walltime"]
    print(f"{(time.time() - start_time) * 1000} ms")

    print("----- Benchmarking Subtracting Columns -----")
    start_time = time.time()
    dataframe["charge"] = dataframe["charge"] - dataframe["walltime"]
    print(f"{(time.time() - start_time) * 1000} ms")

    print("----- Benchmarking Multiplying Columns -----")
    start_time = time.time()
    dataframe["charge"] = dataframe["charge"] * dataframe["walltime"]
    print(f"{(time.time() - start_time) * 1000} ms")



